examples/communication/bluetooth/ReceiveBluetoothFileSample/src/com/digi/xbee/api/receivebluetoothfile/MyBluetoothDataReceiveListener.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019, Digi International Inc.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+package com.digi.xbee.api.receivebluetoothfile;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+
+import com.digi.xbee.api.XBeeDevice;
+import com.digi.xbee.api.exceptions.XBeeException;
+import com.digi.xbee.api.listeners.relay.IBluetoothDataReceiveListener;
+import com.digi.xbee.api.packet.XBeeChecksum;
+import com.digi.xbee.api.utils.ByteUtils;
+
+/**
+ * Class to manage the received data from the Bluetooth interface of the XBee
+ * device.
+ *
+ * <p>Acts as a data listener by implementing the
+ * {@link IBluetoothDataReceiveListener} interface, and is notified when new
+ * data from the Bluetooth interface is received.</p>
+ *
+ * @see IBluetoothDataReceiveListener
+ *
+ */
+public class MyBluetoothDataReceiveListener implements IBluetoothDataReceiveListener {
+
+	// Constants.
+	private static final String MSG_START = "START";
+	private static final String MSG_END = "END";
+	private static final String MSG_ACK = "OK";
+
+	private static final String FILE_NAME = "received_file_%s.txt";
+
+	// Variables.
+	private XBeeDevice device;
+
+	private File receivedFile;
+	private FileOutputStream fileOutputStream;
+
+	/**
+	 * Class constructor. Instantiates a new
+	 * {@code MyBluetoothDataReceiveListener} object for the given XBee device.
+	 *
+	 * @param device XBee device.
+	 */
+	public MyBluetoothDataReceiveListener(XBeeDevice device) {
+		this.device = device;
+	}
+
+	@Override
+	public void dataReceived(byte[] data) {
+		String dataString = new String(data);
+
+		// Check if the data is 'START' or 'END'.
+		if (dataString.equals(MSG_START)) {
+			// If the file exists, remove it.
+			if (receivedFile != null && receivedFile.exists())
+				receivedFile.delete();
+			try {
+				// Create the file and output stream.
+				String dateTime = new SimpleDateFormat("yyyyMMddHHmmss").format(new Date());
```

Comment:
Could the name of the file be included in the START message? This would avoid us to hard code the name of the file and give it the proper extension (user could send a png, pdf or other file type from the mobile app)